src/main/java/jbash/filesystem/FileSystemAPI.java

```diff
@@ -0,0 +1,172 @@
+package jbash.filesystem;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FileSystemAPI {
+    private final Directory root;
+    private Directory currentDirectory;
+
+    private static FileSystemAPI FSAPI = new FileSystemAPI();
+
+    private FileSystemAPI() {
+        this.root = new Directory("root", null);
+        this.currentDirectory = this.root;
+    }
+
+    public static FileSystemAPI getInstance() {
+        return FSAPI;
+    }
+
+    public void reset() {
+        this.FSAPI = new FileSystemAPI();
+    }
+
+    // TODO: Add support for quotes
+    public Optional<FileSystemObject> getFileSystemObject(String path) {
```

Comment:
Javadoc comments for these methods would be really nice to see, especially if FileSystemAPI is front-facing:
https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#format

IntelliJ lets you just type `/**<enter>` above a method signature and it will fill out a boilerplate for you to fill in.

Reply:
im merging the branch ;) but i will include this afterwards. i'll make an issue